backends/verilog/verilog_backend.cc

```diff
@@ -1247,7 +1247,14 @@ void dump_cell(std::ostream &f, std::string indent, RTLIL::Cell *cell)
 	dump_attributes(f, indent, cell->attributes);
 	f << stringf("%s" "%s", indent.c_str(), id(cell->type, false).c_str());
 
-	if (!defparam && cell->parameters.size() > 0) {
+	std::string init;
+	if (cell->name[0] == '$' && reg_ct.count(cell->type) && cell->hasPort("\\Q")) {
```

Comment:
What is the reason for `cell->name[0] == '$' `?

Reply:
I pilfered it from https://github.com/YosysHQ/yosys/blob/e112d2fbf5a31f00ef19e6d05f28fecc1e9c56b9/backends/verilog/verilog_backend.cc#L301

 I imagine it is to save a hash lookup since all internal primitives begin with `$`?

Reply:
I think this should be `cell->type`, otherwise this is checking if cells have an internal name. Although not sure about that other usage, need to look further.

Reply:
Yes, looks like the other usage is to do with renaming internal cell names, hence checking if name is internal rather than an optimisation.

Reply:
Yes indeed, let me fix that up! Wonder why it works at all... [git blame](https://github.com/YosysHQ/yosys/blame/e112d2fbf5a31f00ef19e6d05f28fecc1e9c56b9/backends/verilog/verilog_backend.cc#L433) shows @cliffordwolf

Reply:
No, the existing place with that check is correct I think, because it is part of the logic to prevent internally named signals when `-norename` isn't specified. But that doesn't apply to the usage when looking for init here.

Reply:
Roger that. Removed that first clause entirely now.